07_errors/mohitnag/syncstore.go

```
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"strconv"
+	"sync"
+)
+
+// SyncStore stores Puppy details with Puppy Id as Key and Puppy  as value
+type SyncStore struct {
+	sync.Map
+	m sync.Mutex
+}
+
+// CreatePuppy creates a Puppy in syncstore
+func (s *SyncStore) CreatePuppy(p Puppy) error {
+	s.m.Lock()
+	defer s.m.Unlock()
+	if _, ok := s.Load(p.ID); ok {
+		return ErrorF(Duplicate, "puppy with Id %d already exists", p.ID)
+	}
+	val, _ := strconv.Atoi(p.Value)
```

nit: might be dangerous to ignore the error here.